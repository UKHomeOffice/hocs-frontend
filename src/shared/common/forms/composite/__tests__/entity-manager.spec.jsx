import React from 'react';
import EntityManager from '../entity-manager.jsx';
import { MemoryRouter } from 'react-router-dom';

describe('Entity list component', () => {

    const PAGE = { caseId: '1234', stageId: '5678' };
    const NAME = 'entity_manager';
    const ENTITY = 'entity';
    const BASE_URL = 'http://localhost:8080';
    const MOCK_CALLBACK = jest.fn();

    const DEFAULT_PROPS = {
        page: PAGE,
        name: NAME,
        entity: ENTITY,
        baseUrl: BASE_URL,
    };

    beforeEach(() => {
        MOCK_CALLBACK.mockReset();
    });

    it('should render with default props', () => {
        const WRAPPER = render(<EntityManager baseUrl={'/'}/>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with label when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            label: 'Test entity manager'
        };
        const WRAPPER = render(<EntityManager {...PROPS} />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with additional css when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            className: 'my-css-class'
        };
        const WRAPPER = render(<EntityManager {...PROPS} />);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with remove link when passed in props', () => {
        const choice = value => ({ label: `Choice ${value}`, value: `CHOICE_${value}` });
        const PROPS = {
            ...DEFAULT_PROPS,
            choices: [
                choice('A'),
                choice('B')
            ],
            hasRemoveLink: true
        };
        const WRAPPER = render(
            <MemoryRouter>
                <EntityManager {...PROPS} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with add link when passed in props', () => {
        const PROPS = {
            ...DEFAULT_PROPS,
            hasAddLink: true
        };
        const WRAPPER = render(
            <MemoryRouter>
                <EntityManager {...PROPS} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with download link when passed in props', () => {
        const choice = value => ({ label: `Choice ${value}`, value: `CHOICE_${value}` });
        const PROPS = {
            ...DEFAULT_PROPS,
            choices: [
                choice('A'),
                choice('B')
            ],
            hasDownloadLink: true
        };
        const WRAPPER = render(
            <MemoryRouter>
                <EntityManager {...PROPS} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with tags when passed in props on choice entries', () => {
        const choice = value => ({ label: `Choice ${value}`, value: `CHOICE_${value}`, tags: [value] });
        const PROPS = {
            ...DEFAULT_PROPS,
            choices: [
                choice('A'),
                choice('B')
            ]
        };
        const WRAPPER = render(
            <MemoryRouter>
                <EntityManager {...PROPS} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

    it('should render with created date when passed in props on choice entries', () => {
        const choice = value => ({ label: `Choice ${value}`, value: `CHOICE_${value}`, created: new Date().toString() });
        const PROPS = {
            ...DEFAULT_PROPS,
            choices: [
                choice('A'),
                choice('B')
            ]
        };
        const WRAPPER = render(
            <MemoryRouter>
                <EntityManager {...PROPS} />
            </MemoryRouter>);
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
    });

});