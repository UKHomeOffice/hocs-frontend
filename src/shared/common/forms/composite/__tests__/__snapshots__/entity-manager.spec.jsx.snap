// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity list component should render with add link when passed in props 1`] = `
Array [
  <p
    class="govuk-body"
  >
    No entitys
  </p>,
  <table
    class="govuk-table"
  >
    <tbody
      class="govuk-table__body"
    />
  </table>,
  <a
    class="govuk-body govuk-link"
    href="/http://localhost:8080/entity/add"
  >
    Add entity
  </a>,
]
`;

exports[`Entity list component should render with additional css when passed in props 1`] = `
Array [
  <p
    class="govuk-body"
  >
    No entitys
  </p>,
  <table
    class="govuk-table"
  >
    <tbody
      class="govuk-table__body"
    />
  </table>,
]
`;

exports[`Entity list component should render with created date when passed in props on choice entries 1`] = `
<table
  class="govuk-table"
>
  <tbody
    class="govuk-table__body"
  >
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice A
      </td>
      <td
        class="govuk-table__cell"
      >
        1/1/2020
      </td>
    </tr>
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice B
      </td>
      <td
        class="govuk-table__cell"
      >
        1/1/2020
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Entity list component should render with default props 1`] = `
Array [
  <p
    class="govuk-body"
  >
    No entitys
  </p>,
  <table
    class="govuk-table"
  >
    <tbody
      class="govuk-table__body"
    />
  </table>,
]
`;

exports[`Entity list component should render with download link when passed in props 1`] = `
<table
  class="govuk-table"
>
  <tbody
    class="govuk-table__body"
  >
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice A
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          download="Choice A"
          href="http://localhost:8080/download/entity/CHOICE_A"
        >
          Download
        </a>
      </td>
    </tr>
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice B
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          download="Choice B"
          href="http://localhost:8080/download/entity/CHOICE_B"
        >
          Download
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Entity list component should render with label when passed in props 1`] = `
Array [
  <p
    class="govuk-body"
  >
    No entitys
  </p>,
  <table
    class="govuk-table"
  >
    <caption
      class="govuk-table__caption"
    >
      Test entity manager
    </caption>
    <tbody
      class="govuk-table__body"
    />
  </table>,
]
`;

exports[`Entity list component should render with remove link when passed in props 1`] = `
<table
  class="govuk-table"
>
  <tbody
    class="govuk-table__body"
  >
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice A
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          href="http://localhost:8080/entity/CHOICE_A/remove"
        >
          Remove
        </a>
      </td>
    </tr>
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        Choice B
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          href="http://localhost:8080/entity/CHOICE_B/remove"
        >
          Remove
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Entity list component should render with tags when passed in props on choice entries 1`] = `
<table
  class="govuk-table"
>
  <tbody
    class="govuk-table__body"
  >
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        <strong
          class="govuk-tag margin-right--small"
        >
          A
        </strong>
      </td>
      <td
        class="govuk-table__cell"
      >
        Choice A
      </td>
    </tr>
    <tr
      class="govuk-radios govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        <strong
          class="govuk-tag margin-right--small"
        >
          B
        </strong>
      </td>
      <td
        class="govuk-table__cell"
      >
        Choice B
      </td>
    </tr>
  </tbody>
</table>
`;
